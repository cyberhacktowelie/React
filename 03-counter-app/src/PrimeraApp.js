//FC
// import React from 'react'; //React 17 en adelante no es necesario
// import { Fragment } from "react";

import PropTypes from 'prop-types';

// const PrimeraApp = ( props ) => {
    // const PrimeraApp = ({ saludo = "Hola mundo" }) => {
    const PrimeraApp = ({ saludo, subtitulo='Soy un subtitulo'}) => {

        // const saludo1 = 'Hola mundo';

    //     const objeto = {
    //     nombre: 'Juan',
    //     apellido: 'Perez',
    //     edad: 30
    // };

    // console.log(props);

    return (
        // <Fragment>
        <>
            {/* <h1> {saludo1} </h1> */}
            {/* <h2> {props.saludo} </h2> */}
            <h2>{saludo}</h2>
            {/* <p>Mi PrimeraApp.</p> */}

            {/* <p> {JSON.stringify(objeto)} </p> */}
            {/* <pre> {JSON.stringify(objeto, null, 2)} </pre> */}


            <p>{subtitulo}</p>
        </>
        // </Fragment>
    
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo por defecto'
}

export default PrimeraApp;