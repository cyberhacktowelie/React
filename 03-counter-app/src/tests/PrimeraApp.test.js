/* eslint-disable testing-library/prefer-screen-queries */
// import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";


describe    ('Pruebas en el archivo PrimeraApp.test.js', () => {
    // test('PrimeraApp debe mostrar un saludo', () => {
    //     const saludo = 'Hola soy Goku';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     // console.log(getByText(saludo));

    //     // expect ( getByText(saludo + '!') ).toBeInTheDocument();
    //     expect ( getByText(saludo) ).toBeInTheDocument();

    // });



    test('PrimeraApp debe mostrar un saludo', () => {
        const saludo = 'Hola soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect( wrapper ).toMatchSnapshot();

    });

    test('PrimeraApp debe mostrar un subtitulo enviado por props', () => {
        const saludo = 'Hola soy Goku';
        const subtitulo = 'Soy el subtitulo';
        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo} />);
        expect( wrapper ).toMatchSnapshot();

        const textoParrafo = wrapper.find('p').text();
        // console.log(textoParrafo);

        expect( textoParrafo ).toBe( subtitulo );

    });

});