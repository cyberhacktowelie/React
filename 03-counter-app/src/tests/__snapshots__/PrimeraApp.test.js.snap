// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo PrimeraApp.test.js PrimeraApp debe mostrar un saludo 1`] = `
<Fragment>
  <h2>
    Hola soy Goku
  </h2>
  <p>
    Soy un subtitulo por defecto
  </p>
</Fragment>
`;

exports[`Pruebas en el archivo PrimeraApp.test.js PrimeraApp debe mostrar un subtitulo enviado por props 1`] = `
<Fragment>
  <h2>
    Hola soy Goku
  </h2>
  <p>
    Soy el subtitulo
  </p>
</Fragment>
`;
